File: Array/1_TwoSum.cpp
/******************************************************************************************
1. Two Sum
Easy

Given an array of integers, return indices of the two numbers such that they add up to a specific target.

You may assume that each input would have exactly one solution, and you may not use the same element twice.
*******************************************************************************************/

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int, int> idx;
        
        for(int i=0; i<nums.size(); i++) {
            int x = target - nums[i];
            if(idx.find(x) == idx.end()) {
                idx[nums[i]] = i; 
            }
            else {
                return vector<int> {idx[x], i};
            }
        }
        
        return vector<int> {};
        
    }
};

File: Python/1_Two_Sum.py
"""
1. Two Sum
Easy

Given an array of integers, return indices of the two numbers such that they add up to a specific target.

You may assume that each input would have exactly one solution, and you may not use the same element twice.
"""

class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dict = {}
        for i, val in enumerate(nums):
            if target - val in dict:
                return dict[target - val], i
            else:
                dict[val] = i
                
        return -1, -1
